src/tests/ftest/launch.py

```diff
@@ -1104,336 +966,123 @@ def replace_yaml_file(yaml_file, args, yaml_dir):
         missing_replacements = []
-        display(args, "Modifying contents: {}".format(yaml_file))
+        log.debug("Modifying contents: %s", yaml_file)
         for key in sorted(replacements):
             value = replacements[key]
             if value:
                 # Replace the host entries with their mapped values
-                display(args, "  - Replacing: {} --> {}".format(key, value))
+                log.debug("  - Replacing: %s --> %s", key, value)
                 yaml_data = re.sub(key, value, yaml_data)
             else:
                 # Keep track of any placeholders without a replacement value
-                display(args, "  - Missing:   {}".format(key))
+                log.debug("  - Missing:   %s", key)
                 missing_replacements.append(key)
         if missing_replacements:
             # Report an error for all of the placeholders w/o a replacement
-            log(
-                "Error: Placeholders missing replacements in {}:\n  {}".format(
-                    yaml_file, ", ".join(missing_replacements)))
+            log.error(
+                "Error: Placeholders missing replacements in %s:\n  %s",
+                yaml_file, ", ".join(missing_replacements))
             sys.exit(1)
 
         # Write the modified yaml file into a temporary file.  Use the path to
         # ensure unique yaml files for tests with the same filename.
         orig_yaml_file = yaml_file
         yaml_name = get_test_category(yaml_file)
-        yaml_file = os.path.join(yaml_dir, "{}.yaml".format(yaml_name))
-        log("Creating copy: {}".format(yaml_file))
-        with open(yaml_file, "w") as yaml_buffer:
+        yaml_file = os.path.join(yaml_dir, f"{yaml_name}.yaml")
+        log.debug("Creating copy: %s", yaml_file)
+        with open(yaml_file, "w", encoding="utf-8") as yaml_buffer:
             yaml_buffer.write(yaml_data)
 
-        # Optionally display the file
+        # Optionally display a diff of the yaml file
         if args.verbose > 0:
-            cmd = ["diff", "-y", orig_yaml_file, yaml_file]
-            log(get_output(cmd, False))
+            command = ["diff", "-y", orig_yaml_file, yaml_file]
+            run_local(log, command, check=False)
 
     # Return the untouched or modified yaml file
     return yaml_file
 
 
-def setup_test_directory(args, mode="all"):
+def generate_certs(log):
+    """Generate the certificates for the test.
+
+    Args:
+        log (logger): logger for the messages produced by this method
+
+    Raises:
+        LaunchException: if there is an error generating the certificates
+
+    """
+    log.debug("-" * 80)
+    log.debug("Generating certificates")
+    daos_test_log_dir = os.environ["DAOS_TEST_LOG_DIR"]
+    certs_dir = os.path.join(daos_test_log_dir, "daosCA")
+    run_local(log, ["/usr/bin/rm", "-rf", certs_dir])
+    run_local(log, ["../../../../lib64/daos/certgen/gen_certificates.sh", daos_test_log_dir])
```

Comment:
This relative path makes me uneasy... Could we get the absolute path?

Reply:
Done